From ae2200466c384f478ac25a26b8612c25eab484b9 Mon Sep 17 00:00:00 2001
From: sirse <artiom.tretjakovas2@gmail.com>
Date: Wed, 2 Oct 2024 07:47:54 +0000
Subject: [PATCH] Patch Nim binding

---
 bindings/nim/kzg_abi.nim | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/bindings/nim/kzg_abi.nim b/bindings/nim/kzg_abi.nim
index 0570efd..59e678e 100644
--- a/bindings/nim/kzg_abi.nim
+++ b/bindings/nim/kzg_abi.nim
@@ -20,7 +21,9 @@ when not defined(kzgExternalBlst):
   {.compile: blstPath & "src/server.c"}
   {.passc: "-D__BLST_PORTABLE__"}
 
-{.compile: srcPath & "ckzg.c"}
+{.passl: "-L" & kzgPath & "lib" .}
+{.passl: "-lrust_kzg_arkworks4" .}
+{.passl: "-lm" .}
 
 {.passc: "-I" & escape(bindingsPath) .}
 {.passc: "-I" & escape(srcPath) .}
-- 
2.46.1

