diff --git a/src/c_kzg_4844.c b/src/c_kzg_4844.c
index 847b2c1..5b73b99 100644
--- a/src/c_kzg_4844.c
+++ b/src/c_kzg_4844.c
@@ -792,11 +792,11 @@ static bool pairings_verify(const g1_t *a1, const g2_t *a2, const g1_t *b1, cons
 
 typedef struct { fr_t evals[FIELD_ELEMENTS_PER_BLOB]; } Polynomial;
 
-static void bytes_from_g1(uint8_t out[48], const g1_t *in) {
+void bytes_from_g1(uint8_t out[48], const g1_t *in) {
     blst_p1_compress(out, in);
 }
 
-static C_KZG_RET bytes_to_g1(g1_t* out, const uint8_t bytes[48]) {
+C_KZG_RET bytes_to_g1(g1_t* out, const uint8_t bytes[48]) {
     blst_p1_affine tmp;
     if (blst_p1_uncompress(&tmp, bytes) != BLST_SUCCESS)
         return C_KZG_BADARGS;
