From 8bc31450ffa85ef911550cf35115e89e6ca7b630 Mon Sep 17 00:00:00 2001
From: sirse <artiom.tretjakovas2@gmail.com>
Date: Wed, 2 Oct 2024 06:58:36 +0000
Subject: [PATCH] Patch python bindings

---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index d16c833..e504d12 100644
--- a/setup.py
+++ b/setup.py
@@ -35,10 +35,10 @@ def main():
         ext_modules=[
             Extension(
                 "ckzg",
-                sources=["bindings/python/ckzg_wrap.c", "src/ckzg.c"],
+                sources=["bindings/python/ckzg_wrap.c"],
                 include_dirs=["inc", "src"],
-                library_dirs=["lib"],
-                libraries=["blst"]
+                library_dirs=["lib", "../../target/release"],
+                libraries=[":rust_kzg_blst.a"]
             )
         ],
         cmdclass={
-- 
2.46.1

