From 5b1ad0fd02016130dcfea5a7e84dcf4de2c67a43 Mon Sep 17 00:00:00 2001
From: sirse <artiom.tretjakovas2@gmail.com>
Date: Wed, 23 Oct 2024 11:13:11 +0000
Subject: [PATCH] Patch C# bindings

---
 bindings/csharp/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bindings/csharp/Makefile b/bindings/csharp/Makefile
index df1730c..f115db1 100644
--- a/bindings/csharp/Makefile
+++ b/bindings/csharp/Makefile
@@ -39,7 +39,7 @@ else
 endif
 
 INCLUDE_DIRS = ../../src ../../blst/bindings
-TARGETS = ckzg_wrap.c ../../src/ckzg.c ../../blst/$(BLST_OBJ)
+TARGETS = ckzg_wrap.c
 
 CFLAGS += -O2 -Wall -Wextra -shared
 CFLAGS += ${addprefix -I,${INCLUDE_DIRS}}
@@ -58,7 +58,7 @@ blst:
 
 .PHONY: ckzg
 ckzg: blst
-	$(CLANG_EXECUTABLE) $(CFLAGS) -o $(CKZG_LIBRARY_PATH) $(TARGETS)
+	$(CLANG_EXECUTABLE) $(CFLAGS) -o $(CKZG_LIBRARY_PATH) $(TARGETS) -L ../../lib -lrust_kzg_arkworks -u verify_cell_kzg_proof_batch
 
 .PHONY: ckzg-dotnet
 ckzg-dotnet: ckzg
-- 
2.47.0

