diff --git a/bindings/rust/Cargo.toml b/bindings/rust/Cargo.toml
--- a/bindings/rust/Cargo.toml
+++ b/bindings/rust/Cargo.toml
@@ -1,2 +1,3 @@
+[workspace]
 [package]
 name = "c-kzg"
diff --git a/bindings/rust/build.rs b/bindings/rust/build.rs
--- a/bindings/rust/build.rs
+++ b/bindings/rust/build.rs
@@ -67,10 +67,17 @@ fn main() {
     )
     .unwrap();

+    std::fs::copy(
+        root_dir
+            .join("../../target/release/libblst_from_scratch.a")
+            .as_path(),
+        out_dir.join("libblst_from_scratch.a").as_path(),
+    )
+    .unwrap();
+
     println!("cargo:rustc-link-search={}", out_dir.display());
     println!("cargo:rustc-link-search={}", out_dir.display());
-    println!("cargo:rustc-link-lib=static=ckzg");
-    println!("cargo:rustc-link-lib=static=blst");
+    println!("cargo:rustc-link-lib=blst_from_scratch");

     // Write the compile time variable to a consts.rs file to be imported to the bindings module.
     let const_file = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap()).join("src/consts.rs");
